internal/testing/conformance/run_conformance_test.go

```diff
@@ -0,0 +1,48 @@
+//go:build conformance
+// +build conformance
+
+package conformance
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/consul-api-gateway/internal/testing/e2e"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+)
+
+//This file is intended to help faciliate an easy and local way of running the
+//conformance tests defined here https://github.com/kubernetes-sigs/gateway-api/blob/master/conformance/conformance_test.go
+
+//This is a work in progress
+
+var (
+	testenv   env.Environment
+	hostRoute string
+)
+
+func TestMain(m *testing.M) {
+	//set up test env
+	hostRoute := os.Getenv("DOCKER_HOST_ROUTE")
+	if hostRoute == "" {
+		hostRoute = "host.docker.internal"
+	}
+
+	testenv = env.New()
+	testenv.Setup(
+		SetUpStack(hostRoute),
+	)
+
+	testenv.Finish(
+		e2e.TearDownStack,
+	)
+
+	testenv.Run(m)
+}
+
+func TestRunConformanceTestSuite(t *testing.T) {
+	//TODO wait for gateway to be ready
+
+	//TODO invoke conformance tests
+
+}
```

Comment:
Hey Nathan, I've been playing around with running it that way, pulling down their repo and execing out to the go test. It's kinda working, but its erroring before the test actually runs. I can show you what I mean when we hop on a call, but yes I'd also be interested in how everyone else in the kube-sig chat are running these conformance tests.

Reply:
Hey Nathan, I've been playing around with running it that way, pulling down their repo and execing out to the go test. It's kinda working, but its erroring before the test actually runs. I can show you what I mean when we hop on a call, but yes I'd also be interested in how everyone else in the kube-sig chat are running these conformance tests.

Reply:
From the chatter in the k8s slack, seems like other implementors (i.e. folks like Istio) _are_ essentially just re-writing the upstream `conformance_test.go` and using that after adding in their custom setup logic. TBH wondering if that might be simpler in our case due to the need to patch the upstream manifests -- especially since the suite subpackage accepts an option to [override the base applied manifests](https://github.com/kubernetes-sigs/gateway-api/blob/704c253f29c176ee2e396a5643f4e121018d3b6c/conformance/utils/suite/suite.go#L63).

@nathancoleman @sarahalsmiller what do y'all think about doing something like just duping the upstream `conformance_test.go` and doing something like rendering out the patched manifests via `kustomize` to some temporary file location and then passing in that temporary file to the `BaseManifests` argument there?

Reply:
Hey Andrew, @nathancoleman and I were just talking through this. Essentially I think we like the idea of not having to care what the upstream conformance_test.go looks like, but I also don't want to introduce any additional over head if I don't have to. I'm pretty indifferent either way and I'm happy to implement whichever runner the team thinks is better.

Reply:
So, from reading through some of the evolution of the conformance testing framework, it looks like the sig wants to make it simple for people to actually import the suite as a go module and test it via a simple re-implementation of `conformance_test.go`: https://github.com/kubernetes-sigs/gateway-api/pull/1023

Reply:
The structure of the code and documentation had implied to me that the order of preference was:
1. `make conformance`
2. `go test ./conformance`
3. import the suite, rewrite upstream's `conformance_test.go`

I think the _cleanest_ option is the third if upstream maintainers are okay with that. We'll just need to keep an eye out for shifts in the suite, and I'm not entirely sure they plan on having those be backwards compatible.

Just asked the question in kube-sig Slack [here](https://kubernetes.slack.com/archives/CR0H13KGA/p1648590160555709?thread_ts=1648570950.946279&cid=CR0H13KGA).

Reply:
Definitely feeling better about rewriting `conformance_test.go` now

<img width="388" alt="image" src="https://user-images.githubusercontent.com/3476400/160712616-006662ed-d732-4a57-8928-3b525dd4ada8.png">